### PR TITLE
t4c/client/remote repositories across multiple instances

### PR DESCRIPTION
**Pull request by TimoBbz**
Adds the support of repositories across multiple instances on `t4c/client/remote` docker image, with the environment variable `T4C_JSON`.
See updated doc for more information.

### CODE REVIEW

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]
Tested it, works good on my side.

File: remote/setup_workspace.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
```
[the remark this answers is not in the document]
Comment (TimoBbz) [reply remote/setup_workspace.py:1]: Do you prefer to have separate targets (`run/t4c/client/remote/repositories` and `run/t4c/client/remote/json` for example) or to have the current target overwritten?

File: remote/setup_workspace.py [hunk 2]
```diff
@@ -37,19 +39,45 @@ def replace_config(path: pathlib.Path, key: str, value: str) -> None:
...
+                [repo["repository"] for repo in t4c_repos]
+            ).items()
+            if count > 1
+        ]
+        for repo in t4c_repos:
+            replace_config_repositories(
+                f"{repo['repository']}\\ ({repo['instance']})"
+                if repo["repository"] in duplicate_names
+                else repo["repository"],
+                rf"tcp\://{repo['host']}\:{repo['port']}/{repo['repository']}",
```
Comment (MoritzWeber0) [inline remote/setup_workspace.py:72]: This line is the same for both parts. I'd pass `t4c_host`, `t4c_port` and `repo` to the function.

File: remote/setup_workspace.py [hunk 3]
```diff
@@ -37,19 +39,45 @@ def replace_config(path: pathlib.Path, key: str, value: str) -> None:
     path.write_text(file_content)
 
 
+def replace_config_repositories(key: str, value: str) -> None:
```
Comment (MoritzWeber0) [inline remote/setup_workspace.py:42]:
```suggestion
def inject_t4c_connection_details(name: str, value: str) -> None:
```

I'd choose another name here (describing what the function does). It's not really clear from `key` what it is about. And I'd replace `value`, see the comment below.

File: remote/setup_workspace.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply remote/setup_workspace.py:1]: Looks good. But I'd use `run-t4c/client/remote-legacy` and `run-t4c/client/remote-json`. This makes it consistent with the changes of #52.

[the remark this answers is not in the document]
Comment (TimoBbz) [reply remote/setup_workspace.py:1]: Done.

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]
I'm not able to run the pytest due to a version conflict, I'll try again when support for multiple versions is merged.

**Review by MoritzWeber0 — APPROVED** [review]

**Pull request merged**